### PR TITLE
Remaining defines

### PR DESCRIPTION
**Pull request by jcpunk**
May as well add the remaining defines as squid isnt' too useful without an http_port setup

### CODE REVIEW

**Comment by jcpunk** [comment]
hmmmm I'm not too sure why the tests are failing.... little help?

**Comment by jyaworski** [comment]
@jcpunk can you rebase against the current master?

**Comment by jcpunk** [comment]
re-based, bit suprising on some of the commits that showed up...

**Comment by jyaworski** [comment]
@jcpunk can you look at the tests?

**Comment by jcpunk** [comment]
I have no idea why those are failing.  My ability to produce valid ruby is very limited (and mostly copy/paste).  I'm not really sure what to do with them.

Any pointers in the right direction?

**Comment by traylenator** [comment]
@jcpunk , thanks , I'm back after a week away, let me catch up a bit and I'll take a look.

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -116,4 +116,52 @@
...
+    let :params do
+      { config: '/tmp/squid.conf',
+        snmp_port: { 2000 =>  { 'options' => 'special for 2000' }
+                     },
+
+      }
+    end
+    it { should contain_concat_fragment('squid_header').with_target('/tmp/squid.conf') }
+    it { should contain_concat_fragment('squid_snmp_port_2000').with_content(/^snmp_port\s+2000\s+special for 2000$/) }
+    it { should contain_concat_fragment('squid_snmp_port_2000').with_content(/^if \${process_number} = 3$/) }
```
Comment (traylenator) [inline spec/classes/init_spec.rb]: This fails because indeed process_number has not been set to this test as you have written it will fail.

**Comment by jcpunk** [comment]
I'd swear my most recent branch (about 2 minutes ago) was doing that right, but the tests are still failing....

File: manifests/config.pp [hunk 2]
```diff
@@ -9,6 +9,9 @@
   $workers                       = $squid::workers,
   $acls                          = $squid::acls,
   $http_access                   = $squid::http_access,
+  $http_port                     = $squid::http_port,
```
Comment (traylenator) [inline manifests/config.pp]: For the hash of acl you used a plural parameters *acl* so should be *http_ports* ?

**Comment by jcpunk** [comment]
I've updated the params to be plural.  You're right, it does make more sense that way.

**Comment by jcpunk** [comment]
Any ideas on why the cachedirs doesn't seem to be working?

**Comment by jcpunk** [comment]
I've taken a stab at the rubocop errors

**Comment by jyaworski** [comment]
@jcpunk please rebase.

**Comment by jcpunk** [comment]
For reasons I can't seem to determine, Github doesn't see my new commits in that branch.  So I've Opened a new PR at https://github.com/voxpupuli/puppet-squid/pull/7

**Pull request closed without merging**